tgui/packages/tgui/interfaces/Hypertorus/index.tsx

```diff
@@ -1,51 +1,94 @@
+  power_level: number;
+  selected: string;
+  selectable_fuel: HypertorusFuel[];
+  base_max_temperature: number;
+};
+
+export type HypertorusGas = {
+  id: string;
+  amount: number;
+};
+
+export type HypertorusFilter = {
+  gas_id: string;
+  gas_name: string;
+  enabled: BooleanLike;
+};
+
+export type HypertorusFuel = {
+  id: string | null;
+  requirements: string[];
+  fusion_byproducts: string[];
+  product_gases: string[];
+  recipe_cooling_multiplier: number;
+  recipe_heating_multiplier: number;
+  energy_loss_multiplier: number;
+  fuel_consumption_multiplier: number;
+  gas_production_multiplier: number;
+  temperature_multiplier: number;
+};
 
 const HypertorusMainControls = (props, context) => {
-  const { act, data } = useBackend(context);
+  const { act, data } = useBackend<HypertorusData>(context);
+  const {
+    start_power,
+    start_cooling,
+    start_fuel,
+    start_moderator,
+    power_level,
+    base_max_temperature,
+  } = data;
 
   return (
-    <Section title="Startup">
+    <Section>
       <Stack>
-        <Stack.Item color="label">
+        <Stack.Item color="label" mb={1}>
           {'Start power: '}
           <Button
-            disabled={data.power_level > 0}
-            icon={data.start_power ? 'power-off' : 'times'}
-            content={data.start_power ? 'On' : 'Off'}
-            selected={data.start_power}
+            disabled={power_level > 0}
+            icon={start_power ? 'power-off' : 'times'}
+            content={start_power ? 'On' : 'Off'}
+            selected={start_power}
             onClick={() => act('start_power')}
           />
         </Stack.Item>
         <Stack.Item color="label">
           {'Start cooling: '}
           <Button
             disabled={
-              data.start_fuel === 1 ||
-              data.start_moderator === 1 ||
-              data.start_power === 0 ||
-              (data.start_cooling && data.power_level > 0)
+              start_fuel === 1 ||
+              start_moderator === 1 ||
+              start_power === 0 ||
+              (start_cooling && power_level > 0)
             }
-            icon={data.start_cooling ? 'power-off' : 'times'}
-            content={data.start_cooling ? 'On' : 'Off'}
-            selected={data.start_cooling}
+            icon={start_cooling ? 'power-off' : 'times'}
+            content={start_cooling ? 'On' : 'Off'}
+            selected={start_cooling}
             onClick={() => act('start_cooling')}
           />
         </Stack.Item>
       </Stack>
       <Collapsible title="Recipe selection">
         <HypertorusRecipes
-          baseMaximumTemperature={data.base_max_temperature}
-          enableRecipeSelection={data.power_level === 0}
+          baseMaxTemperature={base_max_temperature}
+          enableRecipeSelection={power_level === 0}
           onRecipe={(id) => act('fuel', { mode: id })}
-          selectableFuels={data.selectable_fuel}
-          selectedFuelID={data.selected}
```

Comment:
this is another bug. for the record, this is bad. there's no reason (that I know of) that you should take backend data then shove it into props. components can call backend data anywhere, so this was just causing it to not update